### PR TITLE
perf graph annotations

### PR DESCRIPTION
**Pull request by psahabu**
- bounded coforall optimization (#6419)
- optimize updateFluff (#6418)
- gcc upgrade to 7.1.0 and reversion to 6.3.0

### CODE REVIEW

File: test/ANNOTATIONS.yaml [hunk 1]
```diff
@@ -128,6 +128,8 @@ all:
   05/05/16:
     - text: ubuntu update to 16.04 (gcc upgraded to 5.3)
       config: shootout
+  06/10/17:
+    - bounded-coforall optimization for remote coforalls (#6419)
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: I would mark this 16 node only

File: test/ANNOTATIONS.yaml [hunk 2]
```diff
@@ -862,6 +876,9 @@ stream-promoted.ml-perf:
   03/21/17:
     - text: Improve const inference for chpl__iterLF (#5736)
       config: 16 node XC
+  06/10/17:
+    - text: bounded-coforall optimization for remote coforalls (#6419)
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: Don't need to re-list this if it's already in the "all" list, right?

File: test/ANNOTATIONS.yaml [hunk 3]
```diff
@@ -242,6 +244,12 @@ all:
   06/08/17:
     - text: upgrade gnu target compiler from 6.3.0 to 7.1.0
       config: Single node XC, 16 node XC
+  06/14/17:
+    - text: upgrade gnu target compiler from 6.3.0 to 7.1.0
+      config: chapcs, chap03, chap04
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: chap03 testing isn't a thing anymore

File: test/ANNOTATIONS.yaml [hunk 4]
```diff
@@ -242,6 +244,12 @@ all:
...
+  06/16/17:
+    - text: revert gnu target compiler from 7.1.0 to 6.3.0
+      config: chapcs
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: Shouldn't the config mirror the above "upgrade" config?

File: test/ANNOTATIONS.yaml [hunk 5]
```diff
@@ -670,6 +678,8 @@ mg-b:
     - Move chpl_getPrivatizedClass() into chpl-privatization.h (#6212)
   05/23/17:
     - Fix NPB MG (#6284)
+  06/13/17:
+    - Optimize updateFluff (#6418)
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: This impacted the prk stencil as well. @benharsh any others that you would expect to be impacted?

Comment (benharsh) [inline test/ANNOTATIONS.yaml]: I don't think so, should just be MG and the Stencil PRK.

File: test/ANNOTATIONS.yaml (hunk 4 above)
Comment (psahabu) [inline test/ANNOTATIONS.yaml]: I thought @dmk42 said it was only reverted on chapcs, but I may have misheard him.

Comment (ronawho) [inline test/ANNOTATIONS.yaml]: setup_gcc_nightly is what gets changed, so all of the chap* and chapcs* machines will get changed at the same time

**Comment by psahabu** [comment]
@ronawho I think I whacked all the goofiness you pointed out

File: test/ANNOTATIONS.yaml [hunk 6]
```diff
@@ -734,6 +745,8 @@ prk-stencil:
     - blockDist for useStencilDist output matrix and more (#3705)
   05/06/16:
     - Problem size increased from --order=1000 to --order=32000 (#3813)
+  06/13/17:
+    - Optimize updateFluff (#6418)
```
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: This is the single-locale graph, and was not impacted by this change

File: test/ANNOTATIONS.yaml (hunk 5 above)
Comment (ronawho) [inline test/ANNOTATIONS.yaml]: @psahabu not on you, but think this graph is misnamed. `mg-b` is the single-node version, and all of these annotations are 16-node so `mg-b` should probably be renamed to `mg.ml-time`

Comment (psahabu) [inline test/ANNOTATIONS.yaml]: I think I fixed this correctly?

**Comment by ronawho** [comment]
I think this is ok. I didn't verify the dates, but seems reasonable

**Pull request merged**